File: workspace-arena/conversor/src/main/xtend-gen/ar/edu/unq/ui/conversor/BasicConverter.java
package ar.edu.unq.ui.conversor;

import org.uqbar.commons.model.annotations.Observable;

@Observable
@SuppressWarnings("all")
public class Conversor {
  private final double factor = 1.60934;
  
  public double milesToKm(final double miles) {
    return (this.factor * miles);
  }
  
  public double kmToMiles(final double km) {
    return (km / this.factor);
  }
}

File: ejemplos/arena/src/main/java/conversor/model/BasicConverter.kt
package conversor.model

import org.uqbar.commons.model.annotations.Observable

@Observable
class BasicConverter {
	private val factor = 1.60934
	var miles: Double = 0.0
	var kilometers: Double = 0.0
	
	fun convert(): Double {
		kilometers = miles * factor
		return kilometers
	}
}
